test/testutil.c

```diff
@@ -145,19 +145,110 @@ int run_tests(const char *test_prog_name)
     return EXIT_SUCCESS;
 }
 
+void fail_message(const char *desc, va_list args, const char *type,
+                         const char *fmt, ...)
+{
+    va_list ap;
+
+    fprintf(stderr, "FAIL ");
+    if (desc != NULL)
+        vfprintf(stderr, desc, args);
+    fprintf(stderr, ": (%s) ", type);
+    va_start(ap, fmt);
+    vfprintf(stderr, fmt, ap);
+    va_end(ap);
+    fputc('\n', stderr);
+}
+
+
+DEFINE_COMPARISONS(int, int, "%d")
+DEFINE_COMPARISONS(unsigned int, uint, "%u")
+
+DEFINE_COMPARISON(void *, ptr, eq, ==, "%p")
+DEFINE_COMPARISON(void *, ptr, ne, !=, "%p")
+
+int ptr_null(const void *p, const char *desc, ...)
+{
+    va_list ap;
+
+    if (p == NULL)
+        return 1;
+    va_start(ap, desc);
+    fail_message(desc, ap, "ptr", "%p == NULL", p);
+    va_end(ap);
+    return 0;
+}
+
+int ptr_not_null(const void *p, const char *desc, ...)
+{
+    va_list ap;
+
+    if (p != NULL)
+        return 1;
+    va_start(ap, desc);
+    fail_message(desc, ap, "ptr", "%p != NULL", p);
+    va_end(ap);
+    return 0;
+}
+
+int bool_true(int b, const char *desc, ...)
+{
+    va_list ap;
+
+    if (b)
+        return 1;
+    va_start(ap, desc);
+    fail_message(desc, ap, "bool", "false == true");
+    va_end(ap);
+    return 0;
+}
+
+int bool_false(int b, const char *desc, ...)
+{
+    va_list ap;
+
+    if (!b)
+        return 1;
+    va_start(ap, desc);
+    fail_message(desc, ap, "bool", "true == false");
+    va_end(ap);
+    return 0;
+}
+
 static const char *print_string_maybe_null(const char *s)
 {
     return s == NULL ? "(NULL)" : s;
 }
 
-int strings_equal(const char *desc, const char *s1, const char *s2)
+int strings_eq(const char *s1, const char *s2, const char *desc, ...)
```

Comment:
A small nit; considering all other comparisons have singular names so far, it seems odd to have plural "strings" here...